rclpy/test/test_executor.py

```diff
@@ -168,6 +168,24 @@ def test_multi_threaded_executor_executes(self):
         finally:
             executor.shutdown()
 
+    def test_multi_threaded_executor_closes_threads(self):
+        self.assertIsNotNone(self.node.handle)
+
+        def get_threads():
+            return {t.name for t in threading.enumerate()}
+
+        main_thread_name = get_threads()
+        # Explicitly specify 2_threads for single thread system failure
+        executor = MultiThreadedExecutor(context=self.context, num_threads=2)
```

Comment:
i added the cosmetic fix here to support the system which has only single thread.